internal/locate/region_cache.go

```diff
@@ -2285,7 +2289,7 @@ func (s *Store) reResolve(c *RegionCache) (bool, error) {
 	storeType := tikvrpc.GetStoreTypeByMeta(store)
 	addr = store.GetAddress()
 	if s.addr != addr || !s.IsSameLabels(store.GetLabels()) {
-		newStore := &Store{storeID: s.storeID, addr: addr, saddr: store.GetStatusAddress(), storeType: storeType, labels: store.GetLabels(), state: uint64(resolved)}
+		newStore := &Store{storeID: s.storeID, addr: addr, peerAddr: store.GetPeerAddress(), saddr: store.GetStatusAddress(), storeType: storeType, labels: store.GetLabels(), state: uint64(resolved)}
```

Comment:
Add peer addr will make MinSafeTS support TiFlash？

Reply:
> Add peer addr will make MinSafeTS support TiFlash？

client-go need to get `StoreSafeTS` from TiFlash by sending GRPC message to `peer addr` of TiFlash. Can see the modification in `updateSafeTS`.